src/black/lines.py

```diff
@@ -268,6 +273,20 @@ def has_magic_trailing_comma(
             return True
 
         if closing.type == token.RSQB:
+            if (
+                Preview.one_tuple_type in self.mode
+                and is_within_annotation(closing)
+                and closing.opening_bracket
+                and is_one_tuple_between(closing.opening_bracket, closing, self.leaves)
+                and closing.parent
+                and closing.parent.prev_sibling
+                and (
+                    list(closing.parent.prev_sibling.leaves())[-1].value
+                    in ("tuple", "Tuple")
```

Comment:
As in anything between the square brackets in a type?

I'm happy with that as it saves us having to check the previous sibling and simplifies things. I think in reality people will likely only add the trailing comma to tuples anyway.

Also it would allow us to also not split on something like: 
```python
class Foo(tuple):
    pass

a: Foo[int,]
```
which is a good thing

Reply:
good point, it's hard to tell that `a = tuple[int,]` is a type alias since it's not a child of `syms.annassign`. Will adjust to look at general one element subscripts 👍

Reply:
I've updated the code to disable magic-trailing-comma for single-element subscripts in general now